ResForge/Classes/EditorManager.swift

```diff
@@ -48,13 +48,47 @@ class EditorManager: RFEditorManager {
         return self.findResource(type: basic ? ResourceType.BasicTemplate : ResourceType.Template, name: type.code)
     }
 
+	private func mappedType(for resource: Resource, forEditor: Bool) -> ResourceType? {
+		do {
+			guard let rmapResource = findResource(type: ResourceType("RMAP"), name: resource.type.code) else { return nil }
+			let reader = BinaryDataReader(rmapResource.data)
+			let destinationType = try reader.readString(length: 4, encoding: .macOSRoman)
+			let editorOnly: UInt16 = try reader.read()
+			var exceptionCount: UInt16 = try reader.read()
+			while exceptionCount > 0 {
+				let exceptionResID: Int16 = try reader.read()
+				let exceptionDestinationType = try reader.readString(length: 4, encoding: .macOSRoman)
+				let exceptionEditorOnly: UInt16 = try reader.read()
+				
+				if resource.id == exceptionResID,
+				   forEditor || exceptionEditorOnly == 0 {
+					return ResourceType(exceptionDestinationType)
+				}
+
+				exceptionCount -= 1
+			}
+			if forEditor || editorOnly == 0 {
+				return ResourceType(destinationType)
+			}
+		} catch {
+		}
+		return nil
+	}
+	
     // MARK: - Protocol functions
 
-    func open(resource: Resource) {
-        if let editor = PluginRegistry.editors[resource.typeCode] {
+	func open(resource: Resource) {
+		open(resource: resource, asType: nil)
+	}
+	
+	private func open(resource: Resource, asType: ResourceType?) {
+		if let editor = PluginRegistry.editors[(asType ?? resource.type).code] {
             self.open(resource: resource, using: editor, template: nil)
-        } else if let template = self.template(for: resource.type) {
-            self.open(resource: resource, using: TemplateEditor.self, template: template)
+		} else if let template = self.template(for: asType ?? resource.type) {
+			self.open(resource: resource, using: TemplateEditor.self, template: template)
+		} else if asType == nil,
+				  let mappedType = mappedType(for: resource, forEditor: true) {
```

Comment:
It looks like the mapped type is only going to kick in if an existing editor/template for the type was not found. Was that the intention, or should it force the type regardless?

Reply:
I thought that safest. ResEdit itself mostly used RMAPs to remap resources to editors for similar resources that knew how to handle both types (something your editor protocol already handles), and for all the various string resource types that exist. So that's what I wanted it to allow as well. HyperCard for example has a bunch of resources that are just renamed versions of 'STR ' for special uses, or a copy of its 'vers' resource so it can internally know its own version even if you built a stack as a Standalone with HC embedded in it and you change the 'vers' to reflect your stack's version.

Reply:
Sure, sounds reasonable.
For reference, for template mappings you can just create a TMPL resource for e.g. `HVRS` with a single field of type `TMPL` named `vers`. This includes the named template resource within the current template.
(It's a new field type I added for ResForge)

Reply:
Interesting! Does TMPL  work for "subclassing" a type? I.e., can I have fields after that field?

I still kinda like RMAP, for one because it's a compatibility thing, and some apps shipped with TMPLs and RMAPs in their resource fork, and because it lets you handle resources like `INTL`, where the format changes based on resource ID.

Reply:
It sure can, checkout some of the templates in the NovaTools plugin for some examples. E.g. the outf and weap templates use it to include the same section in multiple places.
For ID based changes, that's where `KRID` comes in, which is a type of keyed section that selects the case from the resource id. You can see this in action in the NpiL template.